lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -123,22 +132,85 @@ export class PublicClientApplication implements IPublicClientApplication {
         TrustedAuthority.setTrustedAuthoritiesFromConfig(this.config.auth.knownAuthorities, this.config.auth.cloudDiscoveryMetadata);
 
         this.defaultAuthority = null;
-
+        this.broker = false;
         this.initializeBrokering();
     }
 
     private initializeBrokering(): void {
         if (this.config.system.brokerOptions.actAsBroker) {
-            this.broker = new BrokerManager(this.config.system.brokerOptions, this.logger, version);
+            // this.broker = new BrokerManager(this.config.system.brokerOptions, this.logger, version);
+            this.broker = true;
             this.logger.verbose("Acting as Broker");
-            this.broker.listenForMessage();
+            this.listenForBrokerMessage();
         } else if (this.config.system.brokerOptions.allowBrokering) {
-            this.embeddedApp = new BrokerClient(this.config.system.brokerOptions, this.logger, this.config.auth.clientId,  version);
+            this.embeddedApp = new BrokerClient(this.config.system.brokerOptions, this.logger, this.config.auth.clientId, version, this.browserStorage);
             this.logger.verbose("Acting as child");
             this.embeddedApp.initiateHandshake();
         }
     }
 
+    private listenForBrokerMessage(): void {
```

Comment:
Thought we landed on having a `BrokerManager` class that extends `PublicClientApplication`. Did I misunderstand or did that decision change?

Reply:
Just doing things incrementally to see what changes need to be made.